extensions/mapstore/src/main/java/com/hazelcast/mapstore/GenericMapStore.java

```diff
@@ -292,6 +292,7 @@ private int validateColumn(int column, String columnName) {
 
     @Override
     public void destroy() {
+        awaitInitFinished();
```

Comment:
Does it make sense? @frant-hartm

Reply:
Yes it does. The `CREATE MAPPING` is executed from `init` but is async. so a quick create & destroy map could otherwise run the `DROP MAPPING IF EXISTS ...` first and only then execute the `CREATE MAPPING`.